File: Visual Studio LIghtswitch 2012/Chapter16/HelpDeskVB/HelpDeskVB/Client/UserCode/Info.vb
Imports Microsoft.LightSwitch.Threading
Imports System.Windows.Browser
Imports Microsoft.LightSwitch.Security


Namespace LightSwitchApplication

    Public Class Info

        'Listing 16-2. Execution Code for the Logout Button
        Private Sub Logout_Execute()
            Dispatchers.Main.Invoke(
               Sub()
                   HtmlPage.Window.Navigate(
                      New Uri("LogOff.aspx", UriKind.Relative))
               End Sub)
        End Sub

        Private Sub Logout_CanExecute(ByRef result As Boolean)

            Dim logoutCanExecute As Boolean
            Dispatchers.Main.Invoke(
               Sub()
                   logoutCanExecute =
                      (System.Windows.Application.Current.IsRunningOutOfBrowser =
                           False) AndAlso
                       (Application.Current.User.AuthenticationType =
                           AuthenticationType.Forms)
               End Sub)

            result = logoutCanExecute

        End Sub

    End Class

End Namespace

File: Visual Studio LIghtswitch 2012/Chapter16/HelpDeskCS/HelpDeskCS/Client/UserCode/Info.cs
//You can use and redistribute the code from this book (and sample application) for non-commercial and 
//most commercial purposes as long as you acknowledge the source and authorship. 
//You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
//The acknowledgment should include author, title, publisher, and ISBN. 
//An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2012 by Tim Leung, published by Apress, ISBN 9781430250715".

using System;
using System.Linq;
using System.IO;
using System.IO.IsolatedStorage;
using System.Collections.Generic;
using Microsoft.LightSwitch;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Presentation.Extensions;
using Microsoft.LightSwitch.Threading;
using System.Windows.Browser;
using Microsoft.LightSwitch.Security;

namespace LightSwitchApplication
{
    public partial class Info
    {

        //Listing 16-2. Execution Code for the Logout Button
        partial void Logout_Execute()
        {
            Dispatchers.Main.Invoke(() =>
            {
                HtmlPage.Window.Navigate(
                    new Uri("LogOff.aspx", UriKind.Relative));
            });
        }

        partial void Logout_CanExecute(ref bool result)
        {
            bool logoutCanExecute = false;
            Dispatchers.Main.Invoke(() =>
            {
                logoutCanExecute =
                   (System.Windows.Application.Current.IsRunningOutOfBrowser == false)
                      && (Application.Current.User.AuthenticationType ==
                          AuthenticationType.Forms);
            });
            result = logoutCanExecute;
        }

    }
}
